ctapipe/core/container.py

```diff
@@ -59,8 +67,10 @@ def __init__(
         ndim=None,
         allow_none=True,
         max_length=None,
+        default_factory=None,
```

Comment:
since you added `description=` to all containers (in core/container.py), what about using the kwargs-only syntax, to enforce this, here?